### PR TITLE
MULE-12609: Every exported Mule package must contain api

### PR DESCRIPTION
**Pull request by pablokraan**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/internal/enricher/MessageEnricher.java [hunk 1]
```diff
@@ -53,6 +53,7 @@
  * <p/>
  * <b>EIP Reference:</b> <a href="http://eaipatterns.com/DataEnricher.html">http://eaipatterns.com/DataEnricher.html<a/>
  */
+// TODO(pablo.kraan): API - remove org.mule.runtime.core.internal.enricher from exported packages
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/enricher/MessageEnricher.java]: add issue key

File: core/src/main/java/org/mule/runtime/core/internal/util/SecurityUtils.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+ */
+package org.mule.runtime.core.internal.util;
+
+import org.mule.runtime.core.api.config.MuleProperties;
+import org.mule.runtime.core.api.security.tls.TlsConfiguration;
+
+public final class SecurityUtils {
+
+  public static String getSecurityModel() {
+    return System.getProperty(MuleProperties.MULE_SECURITY_SYSTEM_PROPERTY, TlsConfiguration.DEFAULT_SECURITY_MODEL);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/util/SecurityUtils.java]: static imports

File: core/src/main/java/org/mule/runtime/core/internal/util/SecurityUtils.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+  }
+
+  public static boolean isDefaultSecurityModel() {
+    return getSecurityModel().equals(TlsConfiguration.DEFAULT_SECURITY_MODEL);
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/util/SecurityUtils.java]: static imports

File: core/src/main/java/org/mule/runtime/core/internal/util/annotation/AnnotationUtils.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ */
+public class AnnotationUtils {
+
+  public static List<AnnotationMetaData> getMethodAnnotations(Class<?> c, Class<? extends Annotation> ann) {
+    return asList(c.getMethods()).stream().filter(method -> method.isAnnotationPresent(ann))
+        .map(method -> new AnnotationMetaData(c, method, METHOD, method.getAnnotation(ann))).collect(toList());
+  }
+
+  public static <T extends Annotation> Optional<T> getAnnotation(Class<?> aClass, Class<T> annotationType) {
+    return aClass != null ? Optional.ofNullable(aClass.getAnnotation(annotationType)) : Optional.empty();
```
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/internal/util/annotation/AnnotationUtils.java]: static imports

File: core/src/main/resources/META-INF/mule-module.properties [hunk 5]
```diff
@@ -157,7 +150,8 @@ artifact.export.classPackages=org.mule.runtime.core,\
                               org.mule.mvel2.optimizers.dynamic,\
                               org.mule.mvel2.optimizers.impl.refl,\
                               org.mule.mvel2.optimizers.impl.refl.nodes,\
-                              org.mule.mvel2.util
+                              org.mule.mvel2.util,\
+                              org.mule.runtime.core.internal.enricher
```
Comment (elrodro83) [inline core/src/main/resources/META-INF/mule-module.properties]: internal?

Comment (pablokraan) [inline core/src/main/resources/META-INF/mule-module.properties]: I added the export to make tests pass, this is related to the added TODO

**Pull request merged**